src/types/check.rs

```diff
@@ -48,46 +45,25 @@ pub enum ConstTypeError {
     CustomCheckFail(#[from] CustomCheckFailure),
 }
 
-impl PrimType {
-    /// Check that a [`PrimValue`] is a valid instance of this [`PrimType`].
+impl Type {
+    /// Check that a [`Value`] is a valid instance of this [`Type`].
     ///
     /// # Errors
     ///
     /// This function will return an error if there is a type check error.
-    pub fn check_type(&self, val: &PrimValue) -> Result<(), ConstTypeError> {
-        if let PrimType::Alias(alias) = self {
-            return Err(ConstTypeError::NoAliases(alias.name().to_string()));
-        }
-
-        match (self, val) {
-            (PrimType::Extension(e), PrimValue::Extension { c: e_val }) => {
+    pub fn check_type(&self, val: &Value) -> Result<(), ConstTypeError> {
```

Comment:
driveby nit, this should really be called `check_value`. The name `check_type` would be appropriate if this were a method on `Value`